### PR TITLE
Add gru qat int8 test

### PR DESCRIPTION
**Pull request by wozna**
### PR types

Others
### PR changes

Others 
### Describe

This PR adds:
- option to run lexical analysis test with new quantization process. 
- test for GRU QAT int8 model

### CODE REVIEW

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 1]
```diff
@@ -1377,7 +1410,8 @@ if(WITH_IPU)
     SRCS
     ipu_multi_model_profile.cc
     ARGS
-    --model_name="Resnet50"
+    --model_name="Resnet50
+      "
```
Comment (Silv3S) [inline paddle/fluid/inference/tests/api/CMakeLists.txt]:
```suggestion
    --model_name="Resnet50"
```

**Review by Silv3S — DISMISSED** [review]
LGTM

**Review by Silv3S — DISMISSED** [review]

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 2]
```diff
@@ -284,8 +291,9 @@ function(inference_analysis_api_lexical_int8_test_run TARGET_NAME test_binary
     --cpu_num_threads=${CPU_NUM_THREADS_ON_CI}
     --with_accuracy_layer=true
     --use_analysis=true
-    --enable_int8=true
-    --quantized_accuracy=0.01
+    --enable_int8=${enable_int8}
```
Comment (wozna) [inline paddle/fluid/inference/tests/api/CMakeLists.txt]: Correct flag names to make them clearer

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 3]
```diff
@@ -775,12 +783,37 @@ if(WITH_MKLDNN)
...
   inference_analysis_api_lexical_int8_test_run(
-    test_analyzer_lexical_gru_int8 ${LEXICAL_TEST_APP} ${GRU_MODEL_PATH}
-    ${GRU_DATA_PATH} false)
+    test_analyzer_lexical_gru_int8
+    ${LEXICAL_TEST_APP}
+    ${GRU_MODEL_PATH}
+    ${GRU_DATA_PATH}
+    true
+    false
+    false)
```
Comment (tsocha) [inline paddle/fluid/inference/tests/api/CMakeLists.txt]: I'm not a fan of naked bool flags, could you place it in the documenting variable?
You could also use a comment for it.

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 4]
```diff
@@ -775,12 +783,37 @@ if(WITH_MKLDNN)
...
   inference_analysis_api_lexical_int8_test_run(
-    test_analyzer_lexical_gru_int8_multi_gru ${LEXICAL_TEST_APP}
-    ${GRU_MODEL_PATH} ${GRU_DATA_PATH} true)
+    test_analyzer_lexical_gru_int8_multi_gru
+    ${LEXICAL_TEST_APP}
+    ${GRU_MODEL_PATH}
+    ${GRU_DATA_PATH}
+    true
+    false
+    true)
```
Comment (tsocha) [inline paddle/fluid/inference/tests/api/CMakeLists.txt]: as above

File: paddle/fluid/inference/tests/api/CMakeLists.txt [hunk 5]
```diff
@@ -775,12 +783,37 @@ if(WITH_MKLDNN)
...
+                      cf207f8076dcfb8b74d8b6bdddf9090c)
+
+  inference_analysis_api_lexical_int8_test_run(
+    test_analyzer_lexical_gru_qat_int8
+    ${LEXICAL_TEST_APP}
+    "${QAT_GRU_MODEL_DIR}/GRU_quant_acc"
+    ${GRU_DATA_PATH}
+    false
+    true
+    false)
```
Comment (tsocha) [inline paddle/fluid/inference/tests/api/CMakeLists.txt]: as above

**Review by tsocha — CHANGES_REQUESTED** [review]

File: paddle/fluid/inference/tests/api/tester_helper.h [hunk 6]
```diff
@@ -760,15 +764,16 @@ void CompareAccuracy(
     const std::vector<std::vector<PaddleTensor>> &output_slots_quant,
     const std::vector<std::vector<PaddleTensor>> &output_slots_ref,
     int compared_idx) {
-  if ((FLAGS_enable_fp32 && (FLAGS_enable_int8 || FLAGS_enable_bf16)) &&
+  if ((FLAGS_enable_fp32 &&
+       (FLAGS_enable_int8_ptq || FLAGS_enable_int8_qat || FLAGS_enable_bf16)) &&
       (output_slots_quant.size() == 0 || output_slots_ref.size()) == 0)
     throw std::invalid_argument(
         "CompareAccuracy: output_slots vector is empty.");
```
Comment (sfraczek) [inline paddle/fluid/inference/tests/api/tester_helper.h:771]: This if could use some explanation.

**Review by sfraczek — APPROVED** [review]
LGTM

**Review by Silv3S — APPROVED** [review]

**Review by tsocha — APPROVED** [review]

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**